src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -299,7 +299,20 @@ private AcceptTxResult AddCore(Transaction tx, bool isPersistentBroadcast)
                 tx.GasBottleneck = tx.CalculateEffectiveGasPrice(eip1559Enabled, _headInfo.CurrentBaseFee);
                 bool inserted = _transactions.TryInsert(tx.Hash, tx, out Transaction? removed);
                 if (inserted)
-                { 
+                {
+                    Address address = tx.SenderAddress;
+                    if (!_nonces.TryGetValue(address, out AddressNonces? addressNonces))
+                    {
+                        addressNonces = new AddressNonces(tx.Nonce);
+                        _nonces.TryAdd(address, addressNonces);
```

Comment:
Was this missed before? As in, how do `_nonces` got populated before?

Reply:
yes, it was only being populated when it was a ManagedNonce transactions and not in every case.

Reply:
That was the case, and IMO this might not be needed at all. See next comment.